File: source/window/input/Input.java
package window.input;

import imgui.ImGui;

public class Input {
    private Input() {}

    public static boolean isMousePressed(int key) {
        return ImGui.isMouseClicked(key);
    }
    public static boolean isMouseDown(int key) {
        return ImGui.isMouseDown(key);
    }
    public static boolean isMouseReleased(int key) {
        return ImGui.isMouseReleased(key);
    }

    public static boolean isKeyPressed(int key) {
        return ImGui.isKeyPressed(key);
    }
    public static boolean isKeyDown(int key) {
        return ImGui.isKeyDown(key);
    }
    public static boolean isKeyReleased(int key) {
        return ImGui.isKeyReleased(key);
    }

    public static boolean isShiftDown() {
        return ImGui.getIO().getKeyShift();
    }
    public static boolean isCtrlDown() {
        return ImGui.getIO().getKeyCtrl();
    }
    public static boolean isAlDown() {
        return ImGui.getIO().getKeyAlt();
    }
}

File: source/window/input/Input.kt
package window.input

import imgui.ImGui

object Input {
    fun isMousePressed(key: Int): Boolean {
        return ImGui.isMouseClicked(key)
    }

    fun isMouseDown(key: Int): Boolean {
        return ImGui.isMouseDown(key)
    }

    fun isMouseReleased(key: Int): Boolean {
        return ImGui.isMouseReleased(key)
    }

    fun isKeyPressed(key: Int): Boolean {
        return ImGui.isKeyPressed(key)
    }

    fun isKeyDown(key: Int): Boolean {
        return ImGui.isKeyDown(key)
    }

    fun isKeyReleased(key: Int): Boolean {
        return ImGui.isKeyReleased(key)
    }

    val isShiftDown: Boolean
        get() = ImGui.getIO().keyShift
    val isCtrlDown: Boolean
        get() = ImGui.getIO().keyCtrl
    val isAltDown: Boolean
        get() = ImGui.getIO().keyAlt
}